Choice restrictions (#3412)
modelica/ModelicaSpecification: chapters/inheritance.tex

the right of the component; and thus can be redeclared in a consistent
way.

> Having annotations on the \lstinline!redeclare! construct in a modifier is deprecated and the annotations have no effect.

Reviewer: This sounds a little bit like there used to be an affect which is not even mentioned any longer.  Maybe this can be avoided by writing it this way?
```suggestion
Presence of annotations on the \lstinline!redeclare! construct in a modifier is deprecated, but this only affects vendor-specific annotations as none of the annotations in the specification ever had a meaning in this context.
```

Author: I will reformulate it a bit.

Reviewer: OK.  Resolving.

This is a hint for users of the model, and can also be used by the user interface to suggest reasonable redeclaration, where the string comments on the choice declaration can be used as textual explanations of the choices.
The annotation is not restricted to replaceable elements but can also be applied to non-replaceable elements, enumeration types, and simple variables.

> Some choices may be invalid in the current context, e.g., a value might violate the min-attribute.

Reviewer: Formatting:
```suggestion
Some choices may be invalid in the current context, e.g., a value might violate the \lstinline!min!-attribute.
```

Reviewer: The purpose of the sentence also needs to be more clear:
```suggestion
It is allowed to include choices are invalid in some contexts, e.g., a value might violate a \lstinline!min!-attribute.
```

> It is allowed to include choices are invalid in some contexts, e.g., a value might violate a \lstinline!min!-attribute.

Author: ```suggestion
It is allowed to include choices that are invalid in some contexts, e.g., a value might violate a \lstinline!min!-attribute.
```

> \begin{nonnormative}
> Tools may avoid including such choices, mark them as invalid, or detect such violations later.
> \end{nonnormative}

Reviewer: I believe readability would benefit from including the remark in the normative paragraph.
```suggestion
(Options for tools encountering such choices include not showing them, marking them as invalid, or detecting the violations later.)

```
